### PR TITLE
Exclude NodeJS.ReadableStream from compat overloads

### PR DESCRIPTION
**Pull request by hsubox76**
Add NodeJS.ReadableStream to excluded builtin types in the compat overloads script. Maybe there's a more robust way to address this, or can look into it later.

Also fix some type errors in storage.

### CODE REVIEW

File: packages/storage/src/reference.ts [hunk 1]
```diff
@@ -236,7 +236,7 @@ export function getStreamInternal(
 
   ref.storage
     .makeRequestWithTokens(requestInfo, newStreamConnection)
-    .then(stream => stream.pipe(result))
+    .then(stream => (stream as NodeJS.ReadableStream).pipe(result))
```
Comment (Feiyang1) [inline packages/storage/src/reference.ts:239]: Changes in this file seem unrelated?

Comment (hsubox76) [reply packages/storage/src/reference.ts:239]: This is "also fix some type errors in storage", storage was spitting out some errors on build that I was concerned about.

**Review by Feiyang1 — APPROVED** [review]
LGTM. Please fix CI errors.
